tutorial.md

```diff
@@ -0,0 +1,165 @@
+# Boson Functions: A Step By Step Tutorial
+This document will walk you step by step through the process of creating,
+editing, and deploying a Boson Function project.
+
+## Prerequisites
+In order to follow along with this tutorial, you will need to have a few
+tools installed. This project currently is limited to Linux.
+
+<!-- TODO: Add download links -->
+* [Docker](https://docs.docker.com/install/)
+* [oc](https://docs.openshift.com/container-platform/4.2/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands) or [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) CLI
+* [kn](https://knative.dev/docs/install/install-kn/) CLI
+
+
+## Cluster Setup
+To use Boson Functions, you'll need a Kubernetes cluster with Knative Serving
+and Eventing installed. If you have a recent version of OpenShift, you can
+simply install the Serverless Operator. If you don't have a cluster already,
+you can create a simple cluster with [kind](https://kind.sigs.k8s.io/). Follow
+these [step by step instructions](kind-setup.md) to install on your local
+machine.
+
+## Boson Tooling
+The primary interface for Boson project is the `faas` CLI.
+[Download][faas-download] the most recent version and install it some place
+within your `$PATH`.
+
+```sh
+# Be sure to download the correct binary for your operating system
+curl -L -o - faas.gz https://github.com/boson-project/faas/releases/download/v0.6.0/faas_linux_amd64.gz | gunzip > faas && chmod 755 faas
+sudo mv faas /usr/local/bin
+```
+## Configuring a Container Repository
+The unit of deployment in Boson Functions is an [OCI](https://opencontainers.org/)
+container image, typically referred to as a Docker container image.
+
+In order for the `faas` CLI to manage these containers, you'll need to be
+logged in to a container registry.
+
+
+```bash
+docker login -u <username> -p <password> <repository.url>
```

Comment:
is this repository url docker hub?  which i assume is probably the default if not specified or is it the url to a registry that lives inside the openshift/kube cluster?